tests/e2e/e2e_test.go

```diff
@@ -1,14 +1,67 @@
 package e2e
 
 import (
-	"github.com/osmosis-labs/osmosis/v7/tests/e2e/chain"
+	"fmt"
+	"strconv"
+	"time"
 )
 
-func (s *IntegrationTestSuite) TestIBCTokenTransfer() {
+func (s *IntegrationTestSuite) TestSuperfluidVoting() {
 	chainA := s.chainConfigs[0].chain
-	chainB := s.chainConfigs[1].chain
+	s.Run("superfluid_vote_chainA", func() {
+		s.submitSuperfluidProposal(chainA, "gamm/pool/1")
+		s.depositProposal(chainA)
+		s.voteProposal(s.chainConfigs[0])
+		// send gamm tokens to validator's other wallet (non self-delegation wallet)
+		s.sendTx(chainA, 0, "100000000000000000000gamm/pool/1", chainA.Validators[0].PublicAddress, chainA.Validators[0].PublicAddress2)
+		// lock tokens from validator 0 on chain A
+		s.lockTokens(chainA, 0, "100000000000000000000gamm/pool/1", "240s", "val2")
+		// superfluid delegate from validator 0 non self-delegation wallet to validator 1 on chain A
+		s.superfluidDelegate(chainA, s.chainConfigs[0].chain.Validators[1].OperAddress, "val2")
+		// create a text prop, deposit and vote yes
+		s.submitTextProposal(chainA, "superfluid vote overwrite test")
+		s.depositProposal(chainA)
+		s.voteProposal(s.chainConfigs[0])
+		// set delegator vote to no
+		s.voteNoProposal(chainA, 0, "val2")
 
-	// compare coins of reciever pre and post IBC send
-	// diff should only be the amount sent
-	s.sendIBC(chainA, chainB, chainB.Validators[0].PublicAddress, chain.OsmoToken)
+		chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[chainA.ChainMeta.Id][0].GetHostPort("1317/tcp"))
+		sfProposalNumber := strconv.Itoa(chainA.PropNumber)
```

Comment:
We shouldn't store this proposal number on the chain struct, instead, it should be stored in the test suite